home/tests/test_api_v3.py

```diff
@@ -0,0 +1,1026 @@
+    #     if not main_menu:
+    #         main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+    #     parent = main_menu
+
+    #     bodies = [WABody(title, [WABlk(msg_body, document=doc_id_expected)])]
+
+    #     page = PageBuilder.build_cp(
+    #         parent=parent, slug=title.replace(" ", "-"), title=title, bodies=bodies
+    #     )
+    #     response = uclient.get(f"/api/v3/pages/{page.id}/?whatsapp=true")
+    #     content = response.json()
+
+    #     doc_id = content["messages"][0]["document"]
+    #     assert doc_id == page.whatsapp_body._raw_data[0]["value"]["document"]
+
+    # def test_list_items(self, uclient):
+    #     """
+    #     test that list items are present in the whatsapp message
+    #     """
+    #     home_page = HomePage.objects.first()
+    #     main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+    #     target_page = PageBuilder.build_cp(
+    #         parent=main_menu, slug="target-page", title="Target page", bodies=[]
+    #     )
+    #     form = Assessment.objects.create(
+    #         title="Test form", slug="test-form", locale=target_page.locale
+    #     )
+
+    #     page = PageBuilder.build_cp(
+    #         parent=main_menu,
+    #         slug="page",
+    #         title="Page",
+    #         bodies=[
+    #             WABody(
+    #                 "list body",
+    #                 [
+    #                     WABlk(
+    #                         "List message",
+    #                         list_items=[
+    #                             NextListItem("list item 1"),
+    #                             PageListItem("list item 2", page=target_page),
+    #                             FormListItem("list item 3", form=form),
+    #                         ],
+    #                     )
+    #                 ],
+    #             )
+    #         ],
+    #     )
+
+    #     response = uclient.get(f"/api/v3/pages/{page.id}/?whatsapp=true")
+    #     content = response.json()
+
+    #     [item_1, item_2, item_3] = content["body"]["text"]["value"]["list_items"]
+    #     item_1.pop("id")
+    #     item_2.pop("id")
+    #     item_3.pop("id")
+
+    #     assert item_1 == {"type": "item", "value": "list item 1"}
+    #     assert item_2 == {"type": "item", "value": "list item 2"}
+    #     assert item_3 == {"type": "item", "value": "list item 3"}
+
+    #     [item_1, item_2, item_3] = content["body"]["text"]["value"]["list_items_v3"]
+    #     item_1.pop("id")
+    #     item_2.pop("id")
+    #     item_3.pop("id")
+
+    #     assert item_1 == {"type": "next_message", "value": {"title": "list item 1"}}
+    #     assert item_2 == {
+    #         "type": "go_to_page",
+    #         "value": {"title": "list item 2", "page": target_page.id},
+    #     }
+    #     assert item_3 == {
+    #         "type": "go_to_form",
+    #         "value": {"title": "list item 3", "form": form.id},
+    #     }
```

Comment:
There are a lot of full integration tests here, but I think we can simplify the testing a lot of we have one or two integration tests, and then a whole bunch of unit tests, testing things like the edge cases in the serializer logic. But since most of this is commented out, I'm assuming that this is something that will be looked at in the next ticket, so happy for that to take place there.